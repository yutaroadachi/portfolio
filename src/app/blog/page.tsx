import {
  getPersonalBlogList,
  getTechBlogList,
  PersonalBlogList,
  TechBlogList,
} from '@/features/blog/blog-list'
import { ProfileCard } from '@/features/profile/profile-card'
import { generateOpenGraph, generateTitle } from '@/utils/meta/generator'
import { Suspense } from 'react'

const title = 'ブログ'
const description = 'Yutaro Adachi(Frontend Engineer)のブログです。'
export const metadata = {
  title: generateTitle(title),
  description,
  openGraph: generateOpenGraph(title, description),
}

export const revalidate = 300

export default async function BlogListPage() {
  const techBlogListData = getTechBlogList()
  const personalBlogListData = getPersonalBlogList()

  const techBlogList = await techBlogListData

  return (
    <div className="space-y-8">
      <section>
        <h1 id="blog-list">ブログ</h1>
      </section>
      <TechBlogList techBlogList={techBlogList} />
      <Suspense fallback={<h2>Loading…</h2>}>
        {/* @ts-expect-error Server Component */}
        <PersonalBlogListData personalBlogListData={personalBlogListData} />
      </Suspense>
      <ProfileCard />
    </div>
  )
}

const PersonalBlogListData = async ({
  personalBlogListData,
}: {
  personalBlogListData: ReturnType<typeof getPersonalBlogList>
}) => {
  const personalBlogList = await personalBlogListData

  return <PersonalBlogList personalBlogList={personalBlogList} />
}
