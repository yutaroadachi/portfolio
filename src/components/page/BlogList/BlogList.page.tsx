import React, { ComponentProps } from 'react'
import { Layout } from 'src/components/ui/Layout'
import { MetaTag } from 'src/components/ui/MetaTag'
import { BlogList } from './BlogList'

export type BlogListPageProps = ComponentProps<typeof BlogList>

export const BlogListPage = ({ blog }: BlogListPageProps) => {
  return (
    <Layout>
      <MetaTag title="Blog" description="adachiの個人ブログです。" />
      <BlogList blog={blog} />
    </Layout>
  )
}
