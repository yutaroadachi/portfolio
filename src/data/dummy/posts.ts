import { RssFeedItem } from 'src/types/rss'

export const dummyPosts: RssFeedItem[] = Array(13)
  .fill(0)
  .map((_, idx) => ({
    content: '',
    contentSnippet: '',
    creator: '',
    'dc:creator': '',
    enclosure: {
      url: '',
      length: '',
      type: '',
    },
    guid: '',
    isoDate: '',
    link: 'http://example.com',
    pubDate: 'Sat, 28 Aug 2021 00:00:00 GMT',
    title: `テスト記事${idx + 1}`.repeat(idx + 1),
  }))
