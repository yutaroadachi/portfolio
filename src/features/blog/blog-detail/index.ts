export * from './get-personal-blog-detail'
export * from './personal-blog-detail'
