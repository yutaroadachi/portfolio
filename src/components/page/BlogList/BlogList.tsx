import { Box, chakra, SimpleGrid, Stack } from '@chakra-ui/react'
import React from 'react'
import { Biography } from 'src/components/domain/Biography'
import { BlogCard } from 'src/components/domain/BlogCard'
import { PersonalBlogList, TechBlogList } from 'src/types/blog'

export type BlogListProps = {
  techBlog: TechBlogList
  personalBlog: PersonalBlogList
}

export const BlogList = ({ techBlog, personalBlog }: BlogListProps) => {
  return (
    <Box>
      <chakra.h1 id="blog-list" textStyle="h1" color="main" mb={8}>
        ブログ
      </chakra.h1>
      <Stack spacing={8}>
        <TechBlog techBlog={techBlog} />
        <PersonalBlog personalBlog={personalBlog} />
        <Biography />
      </Stack>
    </Box>
  )
}

const TechBlog = ({ techBlog }: Pick<BlogListProps, 'techBlog'>) => {
  return (
    <Box w="full">
      <chakra.h2 id="tech-blog" textStyle="h2WithTreat" color="main" mb={8}>
        テックブログ
      </chakra.h2>
      <SimpleGrid columns={{ base: 2, md: 4 }} spacing={4} w="full">
        {techBlog.map((blog, idx) => {
          return (
            <BlogCard
              href={blog.link}
              title={blog.title}
              publishDate={new Date(blog.pubDate)}
              kind="tech"
              key={idx}
            />
          )
        })}
      </SimpleGrid>
    </Box>
  )
}

const PersonalBlog = ({
  personalBlog,
}: Pick<BlogListProps, 'personalBlog'>) => {
  return (
    <Box w="full">
      <chakra.h2 id="personal-blog" textStyle="h2WithTreat" color="main" mb={8}>
        個人ブログ
      </chakra.h2>
      <SimpleGrid columns={{ base: 2, md: 4 }} spacing={4} w="full">
        {personalBlog.map((blog) => {
          return (
            <BlogCard
              href={`/blog/${blog.id}`}
              title={blog.title}
              publishDate={new Date(blog.publishedAt)}
              kind="personal"
              key={blog.id}
            />
          )
        })}
      </SimpleGrid>
    </Box>
  )
}
