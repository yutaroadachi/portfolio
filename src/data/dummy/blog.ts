import { BlogDetail, BlogList } from 'src/types/blog'

export const dummyBlogList: BlogList = Array(13)
  .fill(0)
  .map((_, idx) => ({
    id: (idx + 1).toString(),
    createdAt: '2021-10-31T00:00:00.0000',
    updatedAt: '2021-10-31T00:00:00.0000',
    publishedAt: '2021-10-31T00:00:00.0000',
    revisedAt: '2021-10-31T00:00:00.0000',
    title: `テスト個人ブログ${idx + 1}`,
    body: `<p>テスト個人ブログ${
      idx + 1
    }です。<br><br><a href="http://example.com" target="_blank" rel="noopener noreferrer">テストリンク</a></p>`,
    description: `テスト個人ブログ${idx + 1}です。`,
  }))

export const dummyBlogDetail: BlogDetail = {
  id: '1',
  createdAt: '2021-10-31T00:00:00.0000',
  updatedAt: '2021-10-31T00:00:00.0000',
  publishedAt: '2021-10-31T00:00:00.0000',
  revisedAt: '2021-10-31T00:00:00.0000',
  title: 'テスト個人ブログ1',
  body: '<p>テスト個人ブログ1です。<br><br><a href="http://example.com" target="_blank" rel="noopener noreferrer">テストリンク</a></p>',
  description: 'テスト個人ブログ1です。',
}
