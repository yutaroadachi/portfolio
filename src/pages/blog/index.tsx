import { GetStaticProps } from 'next'
import React from 'react'
import { BlogList, BlogListProps } from 'src/components/page/BlogList'
import { createGetLayout } from 'src/components/ui/Layout'
import { MetaTag } from 'src/components/ui/MetaTag'
import { getPersonalBlogList } from 'src/lib/microCMS/client'
import { getZennFeed } from 'src/lib/rss/client'

export const getStaticProps: GetStaticProps<BlogListProps> = async () => {
  const [techBlog, personalBlog] = await Promise.all([
    getZennFeed(),
    getPersonalBlogList(),
  ])

  return {
    props: {
      techBlog: techBlog,
      personalBlog: personalBlog,
    },
    revalidate: 5 * 60,
  }
}

export default function BlogListPage({
  techBlog,
  personalBlog,
}: BlogListProps) {
  return (
    <>
      <MetaTag
        title="ブログ"
        description="Yutaro Adachi(ソフトウェアエンジニア)のブログです。"
      />
      <BlogList techBlog={techBlog} personalBlog={personalBlog} />
    </>
  )
}

BlogListPage.getLayout = createGetLayout()
