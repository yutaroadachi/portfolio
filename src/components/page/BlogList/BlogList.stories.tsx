import { Meta, Story } from '@storybook/react'
import React from 'react'
import { dummyBlogList } from 'src/data/dummy/blog'
import { BlogList, BlogListProps } from './BlogList'

export default {
  title: 'page/BlogList',
  component: BlogList,
  args: {
    blog: dummyBlogList,
  },
} as Meta<BlogListProps>

type Template = Story<BlogListProps>

const Template: Template = (args) => <BlogList {...args} />

export const Default: Template = Template.bind({})
