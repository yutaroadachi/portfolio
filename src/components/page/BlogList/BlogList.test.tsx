import React from 'react'
import { dummyBlogList } from 'src/data/dummy/blog'
import { render } from 'src/__tests__/utils'
import { BlogList } from './BlogList'

test('BlogList', () => {
  const { container } = render(<BlogList blog={dummyBlogList} />)
  expect(container).toMatchSnapshot()
})
