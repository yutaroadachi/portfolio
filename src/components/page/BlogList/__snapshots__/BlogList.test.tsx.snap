// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BlogList 1`] = `
<div>
  <h2
    class="css-1yh9w4u"
    id="blog-list"
  >
    Blog
  </h2>
  <div
    class="chakra-stack css-egoftb"
  >
    <div
      class="chakra-linkbox css-w0ckzj"
    >
      <h3
        class="css-0"
      >
        <a
          class="chakra-linkbox__overlay css-1hnz6hu"
          href="/blog/1"
        >
          テスト個人ブログ1
        </a>
      </h3>
      <p
        class="css-1e82jug"
      >
        Posted at 2021/10/31
      </p>
    </div>
    <div
      class="chakra-linkbox css-w0ckzj"
    >
      <h3
        class="css-0"
      >
        <a
          class="chakra-linkbox__overlay css-1hnz6hu"
          href="/blog/2"
        >
          テスト個人ブログ2
        </a>
      </h3>
      <p
        class="css-1e82jug"
      >
        Posted at 2021/10/31
      </p>
    </div>
    <div
      class="chakra-linkbox css-w0ckzj"
    >
      <h3
        class="css-0"
      >
        <a
          class="chakra-linkbox__overlay css-1hnz6hu"
          href="/blog/3"
        >
          テスト個人ブログ3
        </a>
      </h3>
      <p
        class="css-1e82jug"
      >
        Posted at 2021/10/31
      </p>
    </div>
    <div
      class="chakra-linkbox css-w0ckzj"
    >
      <h3
        class="css-0"
      >
        <a
          class="chakra-linkbox__overlay css-1hnz6hu"
          href="/blog/4"
        >
          テスト個人ブログ4
        </a>
      </h3>
      <p
        class="css-1e82jug"
      >
        Posted at 2021/10/31
      </p>
    </div>
    <div
      class="chakra-linkbox css-w0ckzj"
    >
      <h3
        class="css-0"
      >
        <a
          class="chakra-linkbox__overlay css-1hnz6hu"
          href="/blog/5"
        >
          テスト個人ブログ5
        </a>
      </h3>
      <p
        class="css-1e82jug"
      >
        Posted at 2021/10/31
      </p>
    </div>
    <div
      class="chakra-linkbox css-w0ckzj"
    >
      <h3
        class="css-0"
      >
        <a
          class="chakra-linkbox__overlay css-1hnz6hu"
          href="/blog/6"
        >
          テスト個人ブログ6
        </a>
      </h3>
      <p
        class="css-1e82jug"
      >
        Posted at 2021/10/31
      </p>
    </div>
    <div
      class="chakra-linkbox css-w0ckzj"
    >
      <h3
        class="css-0"
      >
        <a
          class="chakra-linkbox__overlay css-1hnz6hu"
          href="/blog/7"
        >
          テスト個人ブログ7
        </a>
      </h3>
      <p
        class="css-1e82jug"
      >
        Posted at 2021/10/31
      </p>
    </div>
    <div
      class="chakra-linkbox css-w0ckzj"
    >
      <h3
        class="css-0"
      >
        <a
          class="chakra-linkbox__overlay css-1hnz6hu"
          href="/blog/8"
        >
          テスト個人ブログ8
        </a>
      </h3>
      <p
        class="css-1e82jug"
      >
        Posted at 2021/10/31
      </p>
    </div>
    <div
      class="chakra-linkbox css-w0ckzj"
    >
      <h3
        class="css-0"
      >
        <a
          class="chakra-linkbox__overlay css-1hnz6hu"
          href="/blog/9"
        >
          テスト個人ブログ9
        </a>
      </h3>
      <p
        class="css-1e82jug"
      >
        Posted at 2021/10/31
      </p>
    </div>
    <div
      class="chakra-linkbox css-w0ckzj"
    >
      <h3
        class="css-0"
      >
        <a
          class="chakra-linkbox__overlay css-1hnz6hu"
          href="/blog/10"
        >
          テスト個人ブログ10
        </a>
      </h3>
      <p
        class="css-1e82jug"
      >
        Posted at 2021/10/31
      </p>
    </div>
    <div
      class="chakra-linkbox css-w0ckzj"
    >
      <h3
        class="css-0"
      >
        <a
          class="chakra-linkbox__overlay css-1hnz6hu"
          href="/blog/11"
        >
          テスト個人ブログ11
        </a>
      </h3>
      <p
        class="css-1e82jug"
      >
        Posted at 2021/10/31
      </p>
    </div>
    <div
      class="chakra-linkbox css-w0ckzj"
    >
      <h3
        class="css-0"
      >
        <a
          class="chakra-linkbox__overlay css-1hnz6hu"
          href="/blog/12"
        >
          テスト個人ブログ12
        </a>
      </h3>
      <p
        class="css-1e82jug"
      >
        Posted at 2021/10/31
      </p>
    </div>
    <div
      class="chakra-linkbox css-w0ckzj"
    >
      <h3
        class="css-0"
      >
        <a
          class="chakra-linkbox__overlay css-1hnz6hu"
          href="/blog/13"
        >
          テスト個人ブログ13
        </a>
      </h3>
      <p
        class="css-1e82jug"
      >
        Posted at 2021/10/31
      </p>
    </div>
  </div>
</div>
`;
